src/tests/math/test_math_arg.f90

```diff
@@ -0,0 +1,41 @@
+program tester
+
+    real :: tol = 1.0e-3
+    call test_math_arg_complex
+    call test_math_argd_complex
+    call test_math_argpi_complex
+    print *, "All tests in `test_math_arg` passed."
+
+contains
+
+    subroutine test_math_arg_complex
+        use stdlib_math, only: arg
+        use stdlib_error, only: check
+
+        call check(arg(2.0*exp((0.0, 0.5))) - 0.5 < tol, msg="arg(2.0*exp((0.0, 0.5))) failed.")
```

Comment:
I haven't looked closely, but here I wonder if there should be an `abs` to prevent the test passing if the function result is substantially less than the target value? Something like  `call check(abs(arg(2.0*exp((0.0, 0.5))) - 0.5) < tol, msg="arg(2.0*exp((0.0, 0.5))) failed.")`

Reply:
Yes, this is my negligence, I will correct it next time. 🤞